### DISCUSSION TITLE
Vue language server code action not work

### DISCUSSION THREAD

**Discussion by vanarok** [helix-editor/helix#9062] [Troubleshooting]
### Summary

An error is printed to the log when I use the built-in vue language server action "Extract into new dumb component".
```
2023-12-12T18:43:27.068 helix_lsp::transport [ERROR] volar <- InternalError: Request codeAction/resolve failed with message: Cannot read properties of undefined (reading 'html')
2023-12-12T18:43:27.069 helix_lsp::transport [ERROR] Tried sending response into a closed channel (id=Num(29)), original request likely timed out
```

### Platform

Linux (Termux)

### Terminal Emulator

Termux

### Installation Method

termux package manager

### Helix Version

helix 23.10 (f6021dd0)

**Comment by the-mikedavis** [comment]
The message "Cannot read properties of undefined (reading 'html')" comes from the language server - it's throwing an error while trying to handle `codeAction/resolve`. This seems like a bug upstream in volar to me but could you post the log while running in verbose mode and/or provide more concrete reproduction steps like an example file?

**Reply by vanarok** [reply]
[helix.log](https://github.com/helix-editor/helix/files/13656962/helix.log)

**Reply by vanarok** [reply]
https://asciinema.org/a/aWTCO1QEVUIzGvH1y0QYTBl5u

**Reply by yowayb** [reply]
@vanarok would you be able to provide a [minimally reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) file?

**Reply by vanarok** [reply]
```
<template>
  <div>Hello <span>world</span></div>
</template>

<script setup></script>
```

**Reply by yowayb** [reply]
I reproduced the error in a fresh project using `npm create vue@latest`.  In VS Code, the code action "Extract into new dumb component" works.  However, according to the mermaid diagram at the bottom of [Vue Language Tools' README](https://github.com/vuejs/language-tools/blob/master/README.md), VS Code's integration with Vue Language Tools is different from all other clients.  I tried to reproduce in Neovim because I figured that would more closely resemble Helix's integration with Vue Language Tools.  Unfortunately, Neovim won't even offer the "Extract into new dumb component" code action.  For this fresh project, in both Helix and Neovim, the LSP logs also include a typescript crash.  However, this typescript crash does not appear in @vanarok 's helix.log

**Reply by yowayb** [reply]
I just discovered how to see the LSP messages between VS Code and Vue Language Tools. To start, open VS Code's Settings and search for "trace server".  You should see "Vue-semantic-server".  You can then set that to "verbose".  Now in the command palette, run "Output: Show Output Channels..." and you should see "Vue Semantic Server".  This channel will show the messages being exchanged.

**Reply by vanarok** [reply]
This problem is reproducible in neovim, so I will create the problem in vue lsp

**Reply by yowayb** [reply]
@vanarok How did you setup Vue Language Tools with Neovim?  I used lazyvim with mason.nvim and this configuration generates an error when I request code actions.

**Reply by yowayb** [reply]
The responses (from the client, to the language server) to `workspace/configuration` (from the language server, to the client) are different (between VS Code and Helix).  VS Code responds with `{ html: true }` while Helix responds with `null`.  This response is sent just before helix receives the error.  Here's the helix log:
```
2023-12-16T18:22:22.886 helix_lsp::transport [INFO] vuels <- {
  "jsonrpc": "2.0",
  "id": 10,
  "method": "workspace/configuration",
  "params": {
    "items": [
      {
        "section": "volar.format.initialIndent"
      }
    ]
  }
}

2023-12-16T18:22:42.886 helix_lsp::transport [INFO] vuels -> {
  "jsonrpc": "2.0",
  "result": [
    null
  ],
  "id": 10
}

2023-12-16T18:22:42.889 helix_lsp::transport [INFO] vuels <- {
  "jsonrpc": "2.0",
  "id": 4,
  "error": {
    "code": -32603,
    "message": "Request codeAction/resolve failed with message: Cannot read properties of undefined (reading 'html')"
  }
}
```
And here's VS Code's logs at the same point:
```
[Trace - 9:23:48 AM] Received request 'workspace/configuration - (19)'.
Params: {
    "items": [
        {
            "section": "volar.format.initialIndent"
        }
    ]
}


[Trace - 9:23:48 AM] Sending response 'workspace/configuration - (19)'. Processing request took 0ms
Result: [
    {
        "html": true
    }
]


[Trace - 9:23:48 AM] Received response 'codeAction/resolve - (82)' in 3ms.
Result: {
    "title": "Extract into new dumb component",
    "data": {
        "uri": "file:///Users/minosh/vue-project-2/src/App.vue",
        "range": [
            286,
            318
        ],
        "newName": "NewComponent"
    },
    "kind": "refactor.move.newFile.dumb",
    "edit": {...TRUNCATED...
```

**Reply by vanarok** [reply]
https://github.com/vuejs/language-tools/issues/3811

**Reply by vanarok** [reply]
you need to install tsdk in config

**Reply by yowayb** [reply]
One thing strange to me (in my helix.log above) is that 20 seconds passes before Helix sends the null result.  20 seconds is also the default timeout for LSP responses, which I assume applies to responses from the language server.

**Reply by yowayb** [reply]
@the-mikedavis since Helix is abiding the LSP spec (returning null to the server in the result array) and @vanarok has opened [an issue with the language server](https://github.com/vuejs/language-tools/issues/3811), I believe this discussion can be closed.
